diff --git a/Blink/Migrator/1860Migration.swift b/Blink/Migrator/1860Migration.swift
index c1c2079b5a961e9c0b4e0917b7f22d0c160b99eb..e7f49b07fe91878632e6f1c512161652cdd31157 100644
--- a/Blink/Migrator/1860Migration.swift
+++ b/Blink/Migrator/1860Migration.swift
@@ -37,7 +37,9 @@ class MigrationStyleFromDefaults: MigrationStep {
   func execute() throws {
     BLKDefaults.loadDefaults()
 
-    let legacy = BLKDefaults.legacyInstance()!
+    guard let legacy = BLKDefaults.legacyInstance() else {
+      return
+    }
     let store = TerminalStyleStore.shared
     let builtIn = TerminalStyle.makeBuiltInDefault()
 
