diff --git a/Blink/Terminal/TermView.m b/Blink/Terminal/TermView.m
index 7ffa417558872490c018540cfa23e469546f84f2..edf6a5cfdee4baac0348a5f42e86037957c54e91 100644
--- a/Blink/Terminal/TermView.m
+++ b/Blink/Terminal/TermView.m
@@ -57,6 +57,8 @@ struct winsize __winSizeFromJSON(NSDictionary *json) {
 @interface TermView () <WKScriptMessageHandler, WKUIDelegate, WKNavigationDelegate, UIGestureRecognizerDelegate, UIEditMenuInteractionDelegate>
 @end
 
+static VSCodeInput *SharedBrowserView;
+
 @implementation TermView {
   WKWebViewGesturesInteraction *_gestureInteraction;
   
@@ -220,6 +222,22 @@ struct winsize __winSizeFromJSON(NSDictionary *json) {
 
 - (void)addBrowserWebView:(NSURL *)url agent: (NSString *)agent injectUIO: (BOOL) injectUIO
 {
+  if (SharedBrowserView) {
+    _browserView = SharedBrowserView;
+    [_browserView removeFromSuperview];
+    [self addSubview:_browserView];
+    _browserView.UIDelegate = self;
+    _browserView.navigationDelegate = self;
+    _browserView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:@[
+      [_browserView.topAnchor constraintEqualToAnchor:_webView.topAnchor],
+      [_browserView.leadingAnchor constraintEqualToAnchor:_webView.leadingAnchor],
+      [_browserView.trailingAnchor constraintEqualToAnchor:_webView.trailingAnchor],
+      [_browserView.bottomAnchor constraintEqualToAnchor:_webView.bottomAnchor]
+    ]];
+    [_browserView loadRequest:[NSURLRequest requestWithURL:url]];
+    return;
+  }
   WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
   if (@available(iOS 18.0, *)) {
     configuration.writingToolsBehavior = UIWritingToolsBehaviorNone;
@@ -241,6 +259,7 @@ struct winsize __winSizeFromJSON(NSDictionary *json) {
 
 
   _browserView = [[VSCodeInput alloc] initWithFrame:CGRectZero configuration:configuration];
+  SharedBrowserView = _browserView;
   _browserView.customUserAgent =
 //  [@"Mozilla/5.0 (Linux; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15 " stringByAppendingString:agent];
   [@"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15 " stringByAppendingString:agent];
@@ -267,6 +286,20 @@ struct winsize __winSizeFromJSON(NSDictionary *json) {
   [_browserView loadRequest:request];
 }
 
+- (void)showBrowserWebView:(NSURL *)url
+{
+  [self addBrowserWebView:url agent:@"" injectUIO:NO];
+  [_device attachInput:_browserView];
+  [_browserView becomeFirstResponder];
+}
+
+- (void)moveSharedBrowserWebViewIfPresent
+{
+  if (SharedBrowserView && SharedBrowserView.superview != self) {
+    [self addBrowserWebView:SharedBrowserView.URL agent:@"" injectUIO:NO];
+  }
+}
+
 - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
   decisionHandler(WKNavigationActionPolicyAllow);
 }
