diff --git a/Blink/Terminal/TermView.h b/Blink/Terminal/TermView.h
index 0124d532ccf2679c38e7109ff0d5ab6ed0759e40..d3296eb3ebb2151cb83718282ec34705db455b3c 100644
--- a/Blink/Terminal/TermView.h
+++ b/Blink/Terminal/TermView.h
@@ -35,7 +35,7 @@
 
 @class TermView;
 @class TermDevice;
-@class TermInput;
+@protocol TermInput;
 @class TermUIState;
 @class LayoutConstraintManager;
 
@@ -46,6 +46,9 @@ extern NSString * TermViewBrowserReadyNotificationKey;
 
 @property BOOL rawMode;
 
+- (void)attachInput:(UIView<TermInput> *)termInput;
+- (void)focus;
+
 - (void)viewIsReady;
 - (void)viewFontSizeChanged:(NSInteger)size;
 - (void)viewWinSizeChanged:(struct winsize)win;
@@ -120,6 +123,8 @@ extern NSString * TermViewBrowserReadyNotificationKey;
 - (void)displayInput:(NSString *)input;
 - (void)apiResponse:(NSString *)name response:(NSString *)response;
 - (void)addBrowserWebView:(NSURL *)url agent: (NSString *)agent injectUIO: (BOOL) injectUIO;
+- (void)showBrowserWebView:(NSURL *)url;
+- (void)moveSharedBrowserWebViewIfPresent;
 
 - (void)modifySideOfSelection;
 - (void)modifySelectionInDirection:(NSString *)direction granularity:(NSString *)granularity;
