diff --git a/Settings/ViewControllers/Appearance/StyleCustomizationView.swift b/Settings/ViewControllers/Appearance/StyleCustomizationView.swift
index 8981444b8efe44a770f2cdb03f3cf5d357d1244b..b203dbc5b298b0416f012ea2402037dadb1798c3 100644
--- a/Settings/ViewControllers/Appearance/StyleCustomizationView.swift
+++ b/Settings/ViewControllers/Appearance/StyleCustomizationView.swift
@@ -266,6 +266,9 @@ struct TerminalPreviewCell: UIViewRepresentable {
 
     var rawMode: Bool = false
 
+    func attachInput(_ termInput: (UIView & TermInput)!) {}
+    func focus() {}
+
     func viewIsReady() {
       guard let tv = termView else { return }
       tv.setWidth(60)
